src/Microsoft.OData.Core/UriParser/Metadata/ODataUriResolver.cs

```diff
@@ -276,47 +276,40 @@ public virtual IEnumerable<IEdmOperationImport> ResolveOperationImports(IEdmMode
         /// <returns>The resolved key list.</returns>
         public virtual IEnumerable<KeyValuePair<string, object>> ResolveKeys(IEdmEntityType type, IDictionary<string, string> namedValues, Func<IEdmTypeReference, string, object> convertFunc)

```

Comment:
I don't think you should change ODataUriResolver. 

You can add a bool to separate your codes and origin codes, or just add AlternateKeyResolver derived from ODataUriResolver to override the ResolveKeys(..) function.